docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+The sub-division of `@Config` classes is accomplished by the addition of `public static` inner classes that act as the physical configuration groupings. Thus, any property that is placed in the inner class is then resolved by the system as not to exist in the default group defined by the `@Config` annotation but rather a new group with the name of the inner class.
+
+Building upon the prior examples, an example usage of property grouping is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property belongs to the default group.")
+  public static boolean myPropertyDefaultGroup = false;
+
+  public static class MyCategory {
+
+    @Comment("This property belongs to the MyCategory group.")
+    public static boolean myPropertyMyCategory = false;
+
+  }
+
+}
+````
+
+
+Alternate Names
+-------------------------
+
+Sometimes, for reasons unknown, we wind up with variable names in a configuration class that would be unsuitable for use as an actual property key. This would be problematic as Forge determines the property key based off of variable names. For this reason, the `@Name` annotation exists.
+
+The name annotation allows for the convenient changing of the property's name from the default value inferred from the name of the variable to something that is deemed more suitable by the modder.
+
+An example usage would be as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property would normally have a horrible name!")
+  @Name("my_well_named_property")
+  public static boolean MY_hoRibbliyNAMEdVairable = false;
+
+}
+````
+
+!! Note
+
+    If you are following conventional Java variable naming practices then the variables defined in your `@Config` class likely contain a great number of capital letters, as seen in the provided example usages (prior to the horribly named one). Since the Forge standard for registration names is all-lowercase, the default property names would therefore be unsuitable, and use of the `@Name` annotation is highly recommended.
+
+    This behavior is partly an oversight and partly a technical limitation. Forge has no intention of punishing any modder who conforms to standard practices. Ideally future releases of Forge will automatically convert traditionally-named variables into conforming property names by replacing capital characters with respective lowercase characters preceded by the underscore character `_`.
```

Comment:
>Forge has no intention of punishing any modder who conforms to standard practices.

Why would they?

Reply:
>Ideally future releases of Forge will automatically

Never gunna happen, your shit is your shit. Stop talking about standards/speculating on Forge and just talk about the functionality.